docs/overview.md

```diff
@@ -56,10 +54,54 @@ model = MyModel(5)
 model.step()
 ```
 
-You should see agents 0-4, activated in random order. See the [tutorial](tutorials/intro_tutorial) or API documentation for more detail on how to add model functionality.
+You should see agents 1-5, activated in random order. See the [tutorial](tutorials/intro_tutorial) or API documentation for more detail on how to add model functionality.
 
 To bootstrap a new model install mesa and run `mesa startproject`
 
+### AgentSet and model.agents
+Mesa 3.0 introduces the AgentSet class and the `model.agents` attribute, providing powerful tools for managing agents.
+
+#### model.agents
+`model.agents` is an AgentSet containing all agents in the model. It's automatically updated when agents are added or removed:
+
+```python
+# Get total number of agents
+num_agents = len(model.agents)
+
+# Iterate over all agents
+for agent in model.agents:
+    print(agent.unique_id)
+```
+
```

Comment:
This would be a place to tell user how to access a single agent from `model.agents`. A sentence with a `model.agents[5]` should be enough.

Reply:
But then you get the 5th agent in the AgentSet. As soon as you shuffle or sort or have multiple agent types this isn't necessarily the agent with unique_id 5 anymore.

It actually never is, since slicing starts at 0 and unique_id at 1.

Reply:
can you just the the agent get function specifying the unique_id?

Reply:
The current way to do this would be:
```Python
model.agents.select(lambda agent: agent.unique_id == 1)[0]
```
Note that the `[0]` converts is from an AgentSet to a single Agent.

It's far from ideal.

I have been pondering this problem for a while. Ideally I would be able to do something like:
```Python
model.agents.select(unique_id>5 and wealth==0)
```
But I haven't found an elegant solution for this.

Reply:
I find `__getitem__` by unique ID more stable and useful than by integer index. The pandas iloc equivalent of the indices gives an illusion of ordering among the agents, which shouldn't be the case. If `__getitem__` becomes `def __getitem__(self, unique_id): return self._agents[unique_id]` as the default, it would simplify the documentation here.

Reply:
Let’s continue this discussion in
- https://github.com/projectmesa/mesa/issues/2316

Reply:
I don't think this is a good place to discuss this idea. I also just note that I object fundamentally to this. It brakes the Sequence protocol. 

In my view, the only defendable use case of `unique_id` is in data collection. I see no other reason for a user to bother with it at all.